From 789c55920c4a3e9fca6606eb5780238a046d51c4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tor=20Arne=20Vestb=C3=B8?= <tor.arne.vestbo@qt.io>
Date: Fri, 3 Jul 2026 14:24:46 +0200
Subject: [PATCH] iOS: Guard against null-focusObject during selection query

When we reset the focus object we might recurse back into callbacks
to canPerformAction:(SEL)action withSender:, where we query the now
null focus object for its selection properties, resulting in an assert.

Pick-to: 6.12 6.11 6.8
Change-Id: I74a4b3543bdb0279fc04cc10fbc6fb395e5ae623
Reviewed-by: Doris Verria <doris.verria@qt.io>
---
 src/plugins/platforms/ios/qiostextinputoverlay.mm | 3 +++
 src/plugins/platforms/ios/qiostextresponder.mm    | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/src/plugins/platforms/ios/qiostextinputoverlay.mm b/src/plugins/platforms/ios/qiostextinputoverlay.mm
index 6fb8e2639e34..b843ed0b1fa0 100644
--- a/src/plugins/platforms/ios/qiostextinputoverlay.mm
+++ b/src/plugins/platforms/ios/qiostextinputoverlay.mm
@@ -32,6 +32,9 @@
 
 static SelectionPair querySelection()
 {
+    if (!QGuiApplication::focusObject())
+        return {};
+
     QInputMethodQueryEvent query(Qt::ImAnchorPosition | Qt::ImCursorPosition);
     QGuiApplication::sendEvent(QGuiApplication::focusObject(), &query);
     int anchorPos = query.value(Qt::ImAnchorPosition).toInt();
diff --git a/src/plugins/platforms/ios/qiostextresponder.mm b/src/plugins/platforms/ios/qiostextresponder.mm
index f189155ddaa6..0286c9ecb3b2 100644
--- a/src/plugins/platforms/ios/qiostextresponder.mm
+++ b/src/plugins/platforms/ios/qiostextresponder.mm
@@ -282,6 +282,9 @@ - (void)sendShortcut:(QKeySequence::StandardKey)standardKey
 
 - (BOOL)hasSelection
 {
+    if (!QGuiApplication::focusObject())
+        return false;
+
     QInputMethodQueryEvent query(Qt::ImAnchorPosition | Qt::ImCursorPosition);
     QGuiApplication::sendEvent(QGuiApplication::focusObject(), &query);
     int anchorPos = query.value(Qt::ImAnchorPosition).toInt();
